eZ/Publish/Core/Repository/Helper/RelationProcessor.php

```diff
@@ -48,21 +51,42 @@ public function __construct(Handler $handler)
      * @param array $locationIdToContentIdMapping An array with Location Ids as keys and corresponding Content Id as values
      * @param \eZ\Publish\SPI\FieldType\FieldType $fieldType
      * @param \eZ\Publish\Core\FieldType\Value $fieldValue Accepted field value.
-     * @param string $fieldDefinitionId
+     * @param \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition $fieldDefinition
+     *
+     * @return \eZ\Publish\SPI\FieldType\ValidationError[]
      */
     public function appendFieldRelations(
         array &$relations,
         array &$locationIdToContentIdMapping,
         SPIFieldType $fieldType,
         BaseValue $fieldValue,
-        $fieldDefinitionId
+        FieldDefinition $fieldDefinition
     ) {
+        $validationErrors = [];
         foreach ($fieldType->getRelations($fieldValue) as $relationType => $destinationIds) {
             if ($relationType === Relation::FIELD) {
-                if (!isset($relations[$relationType][$fieldDefinitionId])) {
-                    $relations[$relationType][$fieldDefinitionId] = array();
+                if (!isset($relations[$relationType][$fieldDefinition->id])) {
+                    $relations[$relationType][$fieldDefinition->id] = array();
+                }
+
+                foreach ($destinationIds as $destinationId) {
+                    try {
+                        $this->persistenceHandler->contentHandler()->loadContentInfo($destinationId);
+                    } catch (NotFoundException $e) {
+                        $validationErrors[] = new ValidationError(
+                            "Relations to Content '%content%' could not be found as part of field '%identifier%'",
```

Comment:
true, and this part I'll probably extract, as far as I could see there is no relation validation anywhere, so that can go in separate story and PR and story and discussion.